src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -932,7 +932,7 @@ export class ConnectionWidget extends lifecycle.Disposable {
 					let value = this.getModelValue(connectionInfo.options[this._customOptions[i].name]);
 					if (value !== '') {
 						if (widget instanceof SelectBox) {
-							widget.selectWithOptionName(value);
+							widget.selectWithOptionName(value?.toString());
```

Comment:
If the type of option is a boolean what about using a checkbox instead?

Reply:
I'll let you make the call here - although I don't think this is the correct fix. The issue is because getModelValue expects a string currently - but the value passed in is defined as an any (it strikes again!)

So the better fix IMO would be to change that to take in any and then call toString() there instead so it's always guaranteed to return a string. (and while you're at it, make sure we aren't assuming anything about the connectionInfo.options type anywhere else here...)

Reply:
It's a good feedback to consider as a design improvement, but it seems this is as per design specs as of now.

Reply:
(let's hope no one ever tries to use an object as a custom option value...)

Reply:
I try to avoid using `any` but it seems it's best option to avoid future problems in this case. Will do that instead.

Reply:
In this case we don't have a choice since the value coming in through connectionInfo is any unfortunately. So we have to be prepared to handle anything.